sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryRowProtoUtils.java

```diff
@@ -0,0 +1,120 @@
+package org.apache.beam.sdk.io.gcp.bigquery;
+
+import static com.google.common.base.Preconditions.checkState;
+
+import com.google.api.services.bigquery.model.TableRow;
+import com.google.cloud.bigquery.v3.RowOuterClass.Row;
+import com.google.cloud.bigquery.v3.RowOuterClass.StructField;
+import com.google.cloud.bigquery.v3.RowOuterClass.StructType;
+import com.google.cloud.bigquery.v3.RowOuterClass.StructValue;
+import com.google.cloud.bigquery.v3.RowOuterClass.Type;
+import com.google.cloud.bigquery.v3.RowOuterClass.Value;
+import com.google.cloud.bigquery.v3.RowOuterClass.Value.ValueCase;
+import com.google.common.collect.ImmutableList;
+import com.google.common.io.BaseEncoding;
+import com.google.protobuf.Timestamp;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+import javax.annotation.Nullable;
+import org.joda.time.format.DateTimeFormat;
+import org.joda.time.format.DateTimeFormatter;
+
+/**
+ * A set of utilities for working with {@link Row} objects. Analogous to {@link BigQueryAvroUtils}.
+ */
+public class BigQueryRowProtoUtils {
+
+  static TableRow convertRowProtoToTableRow(StructType tableSchema, Row row) {
+    return getStructValueAsTableRow(tableSchema, row.getValue());
+  }
+
+  private static TableRow getStructValueAsTableRow(StructType structType, StructValue structValue) {
+    List<StructField> structFieldTypes = structType.getFieldsList();
+    List<Value> structFieldValues = structValue.getFieldsList();
+
+    checkState(structFieldTypes.size() == structFieldValues.size(),
+        "Struct value %s had unexpected field count %d (schema had %d fields)",
+        structValue, structFieldValues.size(), structFieldTypes.size());
+
+    TableRow tableRow = new TableRow();
+    for (int i = 0; i < structFieldTypes.size(); i++) {
+      Object fieldValue = getFieldValueAsTableRowType(
+          structFieldTypes.get(i).getFieldType(), structFieldValues.get(i));
+      if (fieldValue != null) {
+        tableRow.set(structFieldTypes.get(i).getFieldName(), fieldValue);
+      }
+    }
+
+    return tableRow;
+  }
+
+  @Nullable
+  private static Object getFieldValueAsTableRowType(Type fieldType, Value fieldValue) {
```

Comment:
Please make sure this has 100% test coverage.

Reply:
It does (and I've verified as such with a test coverage run under IntelliJ); see RowProtoUtilsTest.java.

Reply:
BigQueryRowProtoUtilsTest.java. Whatever. :-)